### DISCUSSION TITLE
Feature request: External input device buttons mapping

### DISCUSSION THREAD

**Discussion by omadon** [osmandapp/OsmAnd#18673] [Ideas]
Currently key->action mapping is hard coded in the application. It would be nice if there is a configuration screen where this mappings can be changed.

The reason behind this is there are a lot of Bluetooth hardware controllers emulating keyboards and they are all made for specific application with different key mappings. With this option, OsmAnd would be compatible with all of them. If configuration screen is too complicated, external file with  key mappings would also be ok. 

Example: I have DMD2 as main dashboard and OsmAnd  which I use for navigation. DMD2 has zoom in/out button mapped to F6/F7 and OsmAnd has +/- keys. It would be nice If I can also map zoom in/out in OsmAnd to F6/F7.

Key	Device	Action
C	Keyboard	[My location]
D	Keyboard	[Map orientation]
N	Keyboard	Open Navigation view.
S	Keyboard	Open [Search] view.
P	Keyboard	Switch to the next app profile. Show toast like message App profile: Name.
O	Keyboard	Switch to the previous app profile.
↑ 	Keyboard	Move up
↓ 	Keyboard	Move down
← 	Keyboard	Move left
→ 	Keyboard	Move right
\+ 	Keyboard	[Zoom in]
− 	Keyboard	[Zoom out]

**Comment by EugeneZmeuk** [comment]
Hi @omadon 

We added binding keys/actions for external input devices in [OsmAnd 4.6](https://osmand.net/blog/osmand-android-4-6-released/#custom-actions-for-external-controller-buttons). Now you can test it for [the nightly version](https://download.osmand.net/latest-night-build/OsmAnd-android-full-arm64.apk): 

<details>
    <summary>Screenshots</summary>
<table style="width:100%">
  <tr>
    <th></th>
  </tr>
  <tr>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/44466116/addc3081-d9d1-441a-8cf7-b397590ede20"  width="700" height="auto"/> </td>
  </tr>
</table>
</details> 


Next, we will add a choice of additional actions.
